app/models/subscriber.rb

```diff
@@ -0,0 +1,32 @@
+class Subscriber
+  def self.for_discussion(discussion)
+    new(discussion)
+  end
+
+  def initialize(discussion)
+    @discussion = discussion
+    @subscriptions = Subscription.where(discussion_id: discussion.id)
+  end
```

Comment:
Yes, I thought of that, even though it's not exactly the same, since you need a local variable, in the specific case of an initializer it is ok, because you're probably going to create an instance variable anyway.

I was not sure of that, but making the initializer private was mostly as a guideline for future developers. I don't think it is necessary, though.